lib/services/salesforce.rb

```diff
@@ -11,20 +11,28 @@ class AhaServices::Salesforce < AhaService
   
   install_button
 
+  def receive_updated
+    validate_host
+  end
+
   def receive_installed
-    # Validate authentication.
-    client.user_info
-    
-    # Update settings.
-    client.post('/services/apexrest/ahaapp/aha_rest_api/settings', 
-      idea_portal_url: data.idea_portal_url, 
-      jwt_secret_key: data.jwt_secret_key)
-  rescue Exception => e
-    logger.debug("Salesforce authentication problem #{e.class}: #{e.message} #{e.backtrace.join("\n")}")
-    if e.message.include? 'The REST API is not enabled for this Organization.'
-      raise ConfigurationError, "The REST API is not enabled for this Organization."
-    else
-      raise ConfigurationError, "Authentication failed. Please use the 'Authenticate' button to connect to Salesforce."
+    validate_host or raise ConfigurationError, "Please ensure the custom host is a host ending in salesforce.com."
```

Comment:
Agreed. I didn't really like that error message. How about this one? Also, the logger.errors are more visible than the actual raised error I think.
![screen shot 2017-08-22 at 4 31 37 pm](https://user-images.githubusercontent.com/605442/29586137-b203fae6-8757-11e7-99a0-d499df192210.png)